lib/fpm/package/virtualenv.rb

```diff
@@ -0,0 +1,217 @@
+
+        # copy the python project into a temporary one
+        FileUtils.mkdir_p root_folder
+        FileUtils.cp_r path, final_folder
+
+        logger.info( "Updating virtualenv paths ..." )
+
+        # update the virtualenv to point to the installation folder
+        venv = get_virtualenv "#{final_folder}"
+        ::Dir.chdir(venv) do
+            safesystem( "virtualenv-tools --update-path #{dest_path}" )
+        end
+
+        # remove precompiled python files
+        remove_python_compiled_files final_folder
+
+        path = '.'
+        chdir = build_folder
+        source = path
+        destination = "/"
+
+        if attributes[:prefix]
+          destination = File.join(attributes[:prefix], destination)
+        end
+
+        destination = File.join(staging_path, destination)
+
+        logger.info( "Creating source '#{destination}' ..." )
+
+        logger["method"] = "input"
+        begin
+          ::Dir.chdir(chdir) do
+            begin
+              clone(source, destination)
+            rescue Errno::ENOENT => e
+              raise FPM::InvalidPackageConfiguration,
+                "Cannot package the path '#{File.join(chdir, source)}', does it exist?"
+            end
+          end
+        rescue Errno::ENOENT => e
+          raise FPM::InvalidPackageConfiguration,
+            "Cannot chdir to '#{chdir}'. Does it exist?"
+        end
+
+        # Set some defaults. This is useful because other package types
+        # can include license data from themselves (rpms, gems, etc),
+        # but to make sure a simple dir -> rpm works without having
+        # to specify a license.
+        self.license = "unknown"
+        self.vendor = [ENV["USER"], Socket.gethostname].join("@")
+    ensure
+      # Clean up any logger context we added.
+      logger.remove("method")
+      # remove the temporary build folder
+      FileUtils.rm_rf build_folder
+    end
+
+private
+
+    # Delete python precompiled files found in a given folder.
+    def remove_python_compiled_files path
+        Find.find(path) do |path|
+            if path.end_with?'.pyc' or path.end_with?'.pyo'
+                FileUtils.rm path
+            end
+        end
+    end
+
+    # Find the virtualenv installation inside a given path.
+    def get_virtualenv path
+        venv = nil
+        ::Dir["#{path}/*"].each do |file|
+            if File.directory?(file) and File.exist?( "#{file}/bin/activate" )
+                venv = file
+                break
+            end
+        end
+        venv
+    end
+
+    # Copy a file or directory to a destination
+    #
+    # This is special because it respects the full path of the source.
+    # Aditionally, hardlinks will be used instead of copies.
+    #
+    # Example:
+    #
+    #     clone("/tmp/hello/world", "/tmp/example")
+    #
+    # The above will copy, recursively, /tmp/hello/world into
+    # /tmp/example/hello/world
+    def clone(source, destination)
```

Comment:
I've ... see this code before :P 

I think you can subclass Dir instead of copying the code:

```diff
-class FPM::Package::VirtualEnv < FPM::Package
+class FPM::Package::VirtualEnv < FPM::Package::Dir
```